### PR TITLE
inchi: 1.05 -> 1.06

### PR DESCRIPTION
**Pull request by rmcgibbo**
###### Motivation for this change
https://www.inchi-trust.org/downloads/

###### Things done

- [X] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [X] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [X] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

**Comment by r-rmcgibbo** [comment]
Result of `nixpkgs-review pr 116963` at 41ceb637 run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>2 packages skipped due to time constraints:</summary>
  <ul>
    <li>python38Packages.rdkit</li>
    <li>python39Packages.rdkit</li>
  </ul>
</details>
<details>
  <summary>1 package built successfully:</summary>
  <ul>
    <li><a href="https://gist.github.com/09b0a642f7bc5b3fc4e9e7be9a329e50">inchi</a></li></li>
  </ul>
</details>

File: pkgs/development/libraries/inchi/default.nix [hunk 1]
```diff
@@ -8,10 +7,10 @@
 
 stdenv.mkDerivation {
   pname = "inchi";
-  version = "1.05";
+  version = "1.06";
   src = fetchurl {
-    url = "http://www.inchi-trust.org/download/105/INCHI-1-SRC.zip";
-    sha1 = "e3872a46d58cb321a98f4fd4b93a989fb6920b9c";
+    url = "http://www.inchi-trust.org/download/106/INCHI-1-SRC.zip";
```
Comment (SuperSandro2000) [inline pkgs/development/libraries/inchi/default.nix:12]: We should be able to generate that version number by replacing the dot with nothing.

File: pkgs/development/libraries/inchi/default.nix [hunk 2]
```diff
@@ -8,10 +7,10 @@
...
+    sha1 = "z9y3pm78yv4kfhxmg3c78f0ssak9xg09";
```
Comment (SuperSandro2000) [inline pkgs/development/libraries/inchi/default.nix:13]: I think sha1 is adviced to be no longer be used. Can we switch to sha256?

File: pkgs/development/libraries/inchi/default.nix [hunk 3]
```diff
@@ -36,9 +35,9 @@ stdenv.mkDerivation {
     mkdir -p $out/include/inchi
     mkdir -p $doc/share/
 
-    install -m 755 INCHI_API/bin/Linux/libinchi.so.1.05.00 $out/lib
-    ln -s $out/lib/libinchi.so.1.05.00 $out/lib/libinchi.so.1
-    ln -s $out/lib/libinchi.so.1.05.00 $out/lib/libinchi.so
+    install -m 755 INCHI_API/bin/Linux/libinchi.so.1.06.00 $out/lib
+    ln -s $out/lib/libinchi.so.1.06.00 $out/lib/libinchi.so.1
+    ln -s $out/lib/libinchi.so.1.06.00 $out/lib/libinchi.so
```
Comment (SuperSandro2000) [inline pkgs/development/libraries/inchi/default.nix:40]:
```suggestion
    install -m 755 INCHI_API/bin/Linux/libinchi.so.${version}.00 $out/lib
    ln -s $out/lib/libinchi.so.${version}.00 $out/lib/libinchi.so.1
    ln -s $out/lib/libinchi.so.${version}.00 $out/lib/libinchi.so
```

File: pkgs/development/libraries/inchi/default.nix [hunk 4]
```diff
@@ -51,8 +50,8 @@ stdenv.mkDerivation {
   postInstall =
     let
       src-doc = fetchurl {
-        url = "http://www.inchi-trust.org/download/105/INCHI-1-DOC.zip";
-        sha1 = "2f54y0san34v01c215kk0cigzsn76js5";
+        url = "http://www.inchi-trust.org/download/106/INCHI-1-DOC.zip";
+        sha1 = "vkrwg76wh6m5dv9y6igv1q1k2qjpp722";
```
Comment (SuperSandro2000) [inline pkgs/development/libraries/inchi/default.nix:54]: Same as above and this should be moved to a let in before mkDerivation.

**Review by SuperSandro2000 — CHANGES_REQUESTED** [review]

**Pull request merged**